android/PhysicalWeb/app/src/main/java/org/physical_web/physicalweb/Utils.java

```diff
@@ -228,6 +228,49 @@ public static void restoreBlocked(Context context) {
         context).getStringSet("blocked", new HashSet<String>()));
   }
 
+  /**
+   * Checks to see if name matches defined wifi direct structure.
+   * @param context to access shared preferences
+   */
+  public static boolean isWifiDirectName(String name) {
+    String split[] = name.split("-");
+    return (split.length >= 3 && split[0].equals("PW")
+        && isInteger(split[split.length - 1]));
+  }
+
+  /**
+   * Get port from 3 part name.
+   * @param context to access shared preferences
+   */
+  public static int getWifiDirectPort(String name) {
+    String split[] = name.split("-");
+    return  Integer.parseInt(split[split.length - 1]);
+  }
+
+  /**
+   * Get title/url from 3 part name.
+   * @param name of device
+   */
+  public static String getTitleOrURL(String name) {
+    int lastDash = name.lastIndexOf('-');
+    return name.substring(3, lastDash);
+  }
+
+  /**
+   * Check if string represents and integer.
+   * @param string
+   */
+  public static boolean isInteger(String s) {
```

Comment:
You only use this once, so don't bother.  Use a regex instead: `str.matches("\\d+")`